crates/modelardb_compression/src/lib.rs

```diff
@@ -840,58 +811,60 @@ pub mod test_util {
     }
 
     /// Generate constant/random/linear/almost-linear test values with the
-    /// [ThreadRng](rand::rngs::thread::ThreadRng) randomizer. Select the amount of values to be generated
-    /// using `length` and type of values to be generated using [`StructureOfValues`]. If `Random` is
-    /// selected, `min` and `max` is the range of values which can be generated. If `AlmostLinear` is
-    /// selected, `min` and `max` is the maximum and minimum change that should be applied from one value
-    /// to the next. Returns the generated values as a [`Vec`].
+    /// [ThreadRng](rand::rngs::thread::ThreadRng) randomizer. The amount of values to be generated
+    /// will match `timestamps` and their structure will match [`StructureOfValues`]. If `Random` is
+    /// selected, `min` and `max` is the range of values which can be generated. If `AlmostLinear`
+    /// is selected, `min` and `max` is the maximum and minimum change that should be applied from
+    /// one value to the next. Returns the generated values as a [`Vec`].
     pub fn generate_values(
-        length: usize,
+        timestamps: &[i64],
         data_type: StructureOfValues,
         min: Option<f32>,
         max: Option<f32>,
     ) -> Vec<f32> {
-        let mut randomizer = thread_rng();
-        let mut values: Vec<f32> = vec![];
-
         match data_type {
             // Generates almost linear data.
             StructureOfValues::AlmostLinear => {
-                let mut random_linear = vec![];
-                let mut previous_value: f32 = 0.0;
-                for _ in 0..length {
-                    let next_value = (randomizer.sample(Uniform::from(min.unwrap()..max.unwrap())))
-                        + previous_value;
-                    random_linear.push(next_value);
-                    previous_value = next_value;
-                }
-                values.append(&mut random_linear);
+                let a: i64 = thread_rng().gen_range(-10..10);
+                let b: i64 = thread_rng().gen_range(1..50);
+                let mut randomizer = thread_rng();
+
+                timestamps
+                    .iter()
+                    .map(|timestamp| {
+                        (a * timestamp + b) as f32
+                            + randomizer.sample(Uniform::from(min.unwrap()..max.unwrap()))
+                    })
+                    .collect()
             }
-
             // Generates linear data.
             StructureOfValues::Linear => {
-                let mut linear =
-                    Vec::from_iter((10..(length + 1) * 10).step_by(10).map(|v| v as f32));
-                values.append(&mut linear);
+                let a: i64 = thread_rng().gen_range(-10..10);
+                let b: i64 = thread_rng().gen_range(1..50);
+
+                timestamps
+                    .iter()
+                    .map(|timestamp| (a * timestamp + b) as f32)
+                    .collect()
             }
 
```

Comment:
I am not sure what is the correct formatting but there is no newline between the `StructureOfValues::AlmostLinear` and `StructureOfValues::Linear` cases but there is one here. Maybe make this consistent. This also applies to the space between this case and the next.

Reply:
I have removed the additional empty lines.